product_multi_barcode/models/product_barcode.py

```diff
@@ -27,12 +27,14 @@ class ProductBarcode(models.Model):
         compute="_compute_product",
         store=True,
         readonly=False,
+        ondelete="cascade",
     )
     product_tmpl_id = fields.Many2one(
         comodel_name="product.template",
         compute="_compute_product_tmpl",
         store=True,
         readonly=False,
+        ondelete="cascade",
     )
 
     @api.depends("product_id")
```

Comment:
@ilyasProgrammer I think the problem you mention is because this depends does not triggered the recompute as it is missing product_tmpl_id.

```suggestion
    @api.depends("product_id.product_tmpl_id")
```

But, at least your fix is good as it is readonly False, so, could be changed manually.